experimental/pythonreach/main.go

```
@@ -0,0 +1,685 @@
+		if mainEntryRegex.MatchString(line) {
+			hasMainEntry = true
+			return io.EOF // Stop scanning once we find the main entry
+		}
+		return nil
+	})
+
+	if err == io.EOF {
+		return true, nil
+	}
+	return hasMainEntry, err
+}
+
+// findMainEntryPoint scans the target directory for Python files that contain a main entry point.
+func findMainEntryPoint(dir string) ([]string, error) {
+	absDir, err := filepath.Abs(dir)
+	if err != nil {
+		return nil, fmt.Errorf("could not get absolute path for %s: %w", dir, err)
+	}
+	mainFiles := []string{}
+
+	err = filepath.WalkDir(absDir, func(path string, d os.DirEntry, err error) error {
+		if err != nil {
+			return err
+		}
+		if d.IsDir() || !strings.HasSuffix(d.Name(), ".py") {
+			return nil
+		}
+
+		containsEntry, err := fileContainsMainEntryPoint(path)
+		if err != nil {
+			return fmt.Errorf("error reading file %s: %w", path, err)
+		}
+
+		if containsEntry {
+			mainFiles = append(mainFiles, path)
+		}
+
+		return nil
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	return mainFiles, nil
+}
+
+// findManifest searches a directory for a supported manifest file.
+func findManifestFiles(dir string) ([]string, error) {
+	supportedManifests := []string{"poetry.lock"}
+	unsupportedManifests := []string{"requirements.txt", "Pipfile", "Pipefile.lock", "pyproject.toml"}
+	absDir, err := filepath.Abs(dir)
+	if err != nil {
+		return nil, fmt.Errorf("could not get absolute path for %s: %w", dir, err)
+	}
+	manifestFiles := []string{}
+
+	files, err := os.ReadDir(absDir)
+	if err != nil {
+		return nil, fmt.Errorf("could not read directory %s: %w", absDir, err)
+	}
+
+	for _, file := range files {
+		if file.IsDir() {
+			continue
+		}
+		fileName := file.Name()
+		if slices.Contains(supportedManifests, fileName) {
+			manifestFiles = append(manifestFiles, fileName)
+		} else if slices.Contains(unsupportedManifests, fileName) {
+			return nil, fmt.Errorf("unsupported manifest file found: %s", fileName)
```

I think we probably should return an error if any unsupported manifest is found. For example, there may be the case that both requirements.txt and poetry.lock exists in the folder, and error will be returned if requirements.txt is found before poetry.lock. 

I would suggest to just check the supported manifests and we don't need to worry about manifests that we don't support.